packages/ckeditor5-engine/src/model/utils/insertcontent.js

```diff
@@ -226,6 +239,33 @@ class Insertion {
 		this._filterAttributesOf = [];
 	}
 
+	/**
+	 * Updates the last node after the auto paragraphing.
+	 *
+	 * @private
+	 * @param {module:engine/model/node~Node} node The last auto paragraphing node.
+	 */
+	_updateLastNodeFromAutoParagraph( node ) {
+		const positionAfterLastNode = this.writer.createPositionAfter( this._lastNode );
+		const positionAfterNode = this.writer.createPositionAfter( node );
+
+		// If the real end was after the last auto paragraph then update relevant properties.
+		if ( positionAfterNode.isAfter( positionAfterLastNode ) ) {
+			this._lastNode = node;
+
+			/* istanbul ignore if */
+			if ( this.position.parent != node || !this.position.isAtEnd ) {
+				// Algorithm's correctness check. We should never end up here but it's good to know that we did.
+				// At this point the insertion position should be at the end of the last auto paragraph.
+				// Note: This error is documented in other place in this file.
+				throw new CKEditorError( 'insertcontent-invalid-insertion-position', this );
```

Comment:
This is a clone of the previous check that is in `_mergeOnRight()` but this is a slightly different scenario.